program/src/processor.rs

```diff
@@ -435,6 +427,10 @@ fn process_harvest_rewards(program_id: &Pubkey, accounts: &[AccountInfo]) -> Pro
     )?;
     let rewards_to_harvest = eligible_rewards.min(pool_excess_lamports);
 
+    if rewards_to_harvest == 0 {
+        return Err(PaladinRewardsError::NoRewardsToHarvest.into());
```

Comment:
Let's avoid returning an error here -- people likely run scripts which pack as many harvest instructions as possible, and it would be annoying if one of those transaction fails because there's no rewards.

We could add a flag in the instruction data to return an error though if you think this behavior is important -- what do you think?

Reply:
Oops, my bad.I don't think I grokked what you meant by flag at first. I'll update this.

Assuming you're suggesting a parameter like `error_on_no_rewards: bool` - or whatever?

Reply:
Yeah exactly, if you think that's worth it

Reply:
Personally, I think it's a little gratuitous, but there might be a valid use case I'm not seeing. I'm envisioning only a very small subset of people who would use it, and I think a decent client function to query "harvestable" rewards usurps the need for it. Let me know what you think. I'll prepare a commit without it for now.

Reply:
Let's leave out the flag and not return an error then